seerether

high

# user's funds can be transferred without receiving the corresponding dTokens

## Summary
If the mint function fails, there is no mechanism in place to handle the error or revert the transaction. The failure in mint function is not caught by safeTransfer
## Vulnerability Detail
In the userDeposit function, the line of code IDToken(info.dToken).mint(user, dTokenAmount); is responsible for minting dTokens for the user. If there is an issue with the token transfer, such as a failure in the mint function, the funds might not be accounted for correctly.
The vulnerability detail is that if the token transfer fails during the minting process, the user's funds may not be properly reflected in the system. This can lead to a loss of funds or other unexpected behavior, as the user's balance and the total supply of dTokens will not be updated correctly

## Impact
The user's funds could be transferred without receiving the corresponding dTokens, leading to an incorrect balance and potential loss of funds
## Code Snippet
https://github.com/sherlock-audit/2023-06-dodo/blob/main/new-dodo-v3/contracts/DODOV3MM/D3Vault/D3VaultFunding.sol#L41
## Tool used

Manual Review

## Recommendation
Use a try-catch block to catch any exceptions and handle them accordingly
https://github.com/seerether/Dodo/blob/main/Dodomitigate9
